src/converse-service-administration.js

```diff
@@ -0,0 +1,412 @@
+                    'label': __('Announcement')
+                });
+
+                const service_admin_menu = this.el.querySelector('.service-admin-menu');
+                this.collection.forEach(command => {
+                    const itemView = new _converse.ServiceAdminCommandItemView({ model: command });
+                    service_admin_menu.insertAdjacentElement('beforeend', itemView.render());
+                });
+                return this.el;
+            },
+
+            toggleAdminOptions (ev) {
+                ev.preventDefault();
+                u.slideToggleElement(this.el.querySelector(".service-admin-menu"));
+
+                // toggle caret on List-Item
+                const admin_list_caret = this.el.querySelector('.service-admin-selector-caret');
+                if (admin_list_caret.classList.contains('fa-caret-right')) {
+                    admin_list_caret.classList.remove('fa-caret-right');
+                    admin_list_caret.classList.add('fa-caret-down');
+                } else {
+                    admin_list_caret.classList.remove('fa-caret-down');
+                    admin_list_caret.classList.add('fa-caret-right');
+                }
+            }
+        });
+
+        _converse.ServiceAdminBox = _converse.ChatBox.extend({
+            defaults () {
+                return {
+                    'bookmarked': false,
+                    'hidden': ['mobile', 'fullscreen'].includes(_converse.view_mode),
+                    'message_type': 'service-administration',
+                    'num_unread': 0,
+                    'time_opened': this.get('time_opened') || (new Date()).getTime(),
+                    'type': _converse.SERVICE_ADMIN_TYPE,
+                }
+            },
+
+            initialize () {
+                this.set({'box_id': `box-${btoa(this.get('jid'))}`});
+            }
+        });
+
+        _converse.ServiceAdminBoxView = _converse.ChatBoxView.extend({
+            className: 'chatbox',
+            id: 'service-administration',
+            events: {
+                'click .send-button': 'sendMessageToServer',
+                'click .chatbox-btn': 'close'
+            },
+
+            initialize () {
+                _converse.api.trigger('chatBoxInitialized', this);
+                this.listenTo(this.model, 'destroy', this.hide);
+                this.listenTo(this.model, 'hide', this.hide);
+                this.listenTo(this.model, 'show', this.show);
+                this.listenTo(this.model, 'change:closed', this.ensureClosedState);
+
+                this.render();
+            },
+
+            render () {
+                this.el.innerHTML = tpl_service_admin_chat({
+                    'service_admin_header': this.model.attributes.header_text,
+                    'info_close': __('Close'),
+                    'send_button_text': this.model.attributes.send_button_text,
+                    'placeholder_text': this.model.attributes.placeholder_text,
```

Comment:
You can use `this.model.getJSON()`, then you don't have to type fields out that re-use the same name.

For example:
```
this.el.innerHTML = tpl_service_admin_chat(Object.assign(this.model.toJSON(), {
    'service_admin_header': this.model.get('header_text'),
    'info_close': __('Close'),
});
```